pages/cql/cms124_FHIR-7.2.000.cql

```diff
@@ -0,0 +1,127 @@
+
+parameter "Measurement Period" Interval<DateTime>
+
+context Patient
+
+define "Ethnicity Categories":
+		flatten (
+        (
+            Patient.extension Extensions
+                where Extensions.url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity'
+        ) EthnicityExtensions
+            where EthnicityExtensions.extension.url = 'ombCategory'
+        return EthnicityExtensions.extension
+		)
+
+define "SDE Ethnicity":
+		"Ethnicity Categories" EthnicityExt
+				where FHIRHelpers.ToCode(EthnicityExt.value as FHIR.Coding) in "Ethnicity"
+		return FHIRHelpers.ToCode(EthnicityExt.value as FHIR.Coding).display
+
+		// DT: Not addressed in QI Core / FHIR
+// CRS: Use the "Payer" ValueSet to investigate what codes are expected and investigate whether an approach can be used with Coverage resource
+define "SDE Payer":
+	["Patient Characteristic Payer": "Payer"]
+
+define "Race Categories":
+		flatten (
+        (
+            Patient.extension Extensions
+                where Extensions.url = 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-race'
+        ) RaceExtensions
+            where RaceExtensions.extension.url = 'ombCategory'
+        return RaceExtensions.extension
+    )
+
+define "SDE Race":
+		"Race Categories" RaceExt
+				where FHIRHelpers.ToCode(RaceExt.value as FHIR.Coding) in "Race"
+		return FHIRHelpers.ToCode(RaceExt.value as FHIR.Coding).display
+		
+define "SDE Sex":
+		[Patient: gender in "ONC Administrative Sex"] P
+			return P.gender
+
+define "Numerator":
+	exists "Pap Test Within 3 Years"
+		or exists "Pap Test With HPV Within 5 Years"
+
+define "Denominator":
+	"Initial Population"
+
+define "Pap Test with Results":
+	[Observation: "Pap Test"] PapTest
+		where PapTest.value is not null
+
+// DT need to investigate more
+define "Pap Test Within 3 Years":
+	"Pap Test with Results" PapTest
+		where PapTest.relevantPeriod 3 years or less before end of "Measurement Period"
+
+define "Qualifying Encounters":
+	( [Encounter: "Office Visit"]
+		union [Encounter: "Preventive Care Services - Established Office Visit, 18 and Up"]
+		union [Encounter: "Preventive Care Services-Initial Office Visit, 18 and Up"]
+		union [Encounter: "Home Healthcare Services"] ) ValidEncounter
+		where ValidEncounter.period during "Measurement Period"
+
+define "PapTest Within 5 Years":
+	( "Pap Test with Results" PapTestOver30YearsOld
+			with ["Patient Characteristic Birthdate"] BirthDate
+				such that Global."CalendarAgeInYearsAt"(BirthDate.birthDatetime, start of PapTestOver30YearsOld.relevantPeriod)>= 30
+					and PapTestOver30YearsOld.relevantPeriod 5 years or less before end of "Measurement Period"
+	)
+
+define "Pap Test With HPV Within 5 Years":
+	"PapTest Within 5 Years" PapTestOver30YearsOld
+		with [Observation: "HPV Test"] HPVTest
+			such that HPVTest.value is not null
+				and ( HPVTest.effective starts 1 day or less before or on day of start of PapTestOver30YearsOld.relevantPeriod

```

Comment:
Add `and HPVTest in { 'final', 'amended', 'corrected' }`. Also, relevantPeriod is a QDM element. Use FHIR Observation.effective element as discussed in mapping tables.